### PR TITLE
Clarify release process documentation.

### PR DESCRIPTION
**Pull request by lbakerchef**
### Description

See title.

### CODE REVIEW

File: RELEASE_PROCESS.md [hunk 1]
```diff
@@ -49,30 +49,37 @@ Starting download https://packages.chef.io/files/current/chef-server/14.2.23/ubu
 
 ### Testing the Release
 
-Every merge to chef-server master must be built, and this build must be tested with the full Umbrella automated integration test pipeline at https://buildkite.com/chef/chef-umbrella-master-chef-server-full.
+Every merge to chef-server master must be built, and this build must be tested with the full Umbrella automated integration test pipeline at https://buildkite.com/chef/chef-umbrella-master-chef-server-full (NOTE: use https://buildkite.com/chef/chef-umbrella-master-chef-server for testing the 12.17.15 upgrade path).
```
Comment (PrajaktaPurohit) [inline RELEASE_PROCESS.md:52]: Every merge to master automatically runs through adhoc and we run the umbrella nightly for all the changes made in that day.

File: RELEASE_PROCESS.md [hunk 2]
```diff
@@ -49,30 +49,37 @@ Starting download https://packages.chef.io/files/current/chef-server/14.2.23/ubu
...
 Umbrella Testing Step-by-Step:
 
-1. Navigate your web browser to https://buildkite.com/chef/chef-umbrella-master-chef-server
+1. Navigate your web browser to https://buildkite.com/chef/chef-umbrella-master-chef-server-full
 1. Select 'New Build'.
 1. Leave 'Branch' set to 'master'.
 1. Select 'Options' to expand the 'Environment Variables' field.
-1. Enter `INSTALL_VERSION=<version number>` into the 'Options | Environment Variables' field, where _version number_ is the stable  version number of the release you wish to test upgrading FROM (for example 14.0.65).
-1. Enter `UPGRADE_VERSION=<version number>` into the 'Options | Environment Variables' field, where _version number_ is the current version number of the release candidate you wish to test upgrading TO (for example 14.2.2).
+1. Enter `INSTALL_VERSION=<version number>` into the 'Options | Environment Variables' field, where _version number_ is the stable channel  version number of the release you wish to test upgrading FROM (for example 14.0.65).
```
Comment (PrajaktaPurohit) [inline RELEASE_PROCESS.md:72]: might want to update the numbering on these?

File: RELEASE_PROCESS.md (hunk 1 above)
Comment (lbakerchef) [reply RELEASE_PROCESS.md:52]: I added this in the form of a NOTE.

File: RELEASE_PROCESS.md (hunk 2 above)
Comment (lbakerchef) [reply RELEASE_PROCESS.md:72]: I updated it to reflect the 12.17.15 -> YOUR-RELEASE path.

**Review by PrajaktaPurohit — APPROVED** [review]

**Pull request merged**